internal/apiclient/pricing.go

```diff
@@ -93,6 +118,82 @@ func NewPricingAPIClient(ctx *config.RunContext) *PricingAPIClient {
 		Currency:       currency,
 		EventsDisabled: ctx.Config.EventsDisabled,
 	}
+
+	initCache(ctx, c)
+	pricingClient = c
+	return c
+}
+
+func initCache(ctx *config.RunContext, c *PricingAPIClient) {
+	if ctx.Config.PricingCacheDisabled {
+		return
+	}
+
+	cacheFile := filepath.Join(ctx.Config.CachePath(), config.InfracostDir, "pricing.gob")
+	c.cacheFile = cacheFile
+	cache := loadCacheFromFile(cacheFile)
+	c.cache = &cache
+
+	c.mu = &sync.RWMutex{}
+	c.objectLimit = 200
+	if ctx.Config.PricingCacheObjectSize > 0 {
+		c.objectLimit = ctx.Config.PricingCacheObjectSize
+	}
+}
+
+func loadCacheFromFile(cacheFile string) map[uint64]gjson.Result {
+	cache := make(map[uint64]gjson.Result)
+	info, err := os.Stat(cacheFile)
+	if err != nil {
+		return cache
+	}
+
+	// if the cache is older than a day don't use it
+	if info.ModTime().Before(time.Now().AddDate(0, 0, -1)) {
+		return cache
+	}
```

Comment:
Will this mean that if a user continuously uses cached queries every 20 hours then they never expire since we load them from the cache and write them to a new cache file each time?

Reply:
yup makes sense, we've disccuseed and I'm changing the Implementation